### PR TITLE
Add EventTriggersFunctionOperator to Kafka Integrations page

### PR DESCRIPTION
**Pull request by TJaniF**
New operator in the Kafka provider, adding the description of the parameters to the How it works section (the Kafka provider doesn't have dedicated documentation outside of this integrations page right now). This operator will most likely be used in the upcoming blogpost.
Please tell me if my explanation makes sense, basically the operator indefinitely re-defers itself and kicks off a function for every time its conditions are met, so it is a bit of a 🦄 .

Also fixes a broken link. 👀

### CODE REVIEW

File: learn/airflow-kafka.md [hunk 1]
```diff
@@ -320,6 +321,19 @@ The AwaitKafkaMessageOperator is a [deferrable operator]((https://docs.astronome
 - `poll_interval`: The amonut of time in seconds that the task should wait in the deferred state.
 - `xcom_push_key`: The key under which to save the returned data to XCom.
 
+### EventTriggersFunctionOperator
+
+The EventTriggerFunctionOperator is a [deferrable operator](https://docs.astronomer.io/learn/deferrable-operators) that will wait for a specific message to be published to one or more Kafka topics, similar to the AwaitKafkaMessageOperator. After consuming a message that fulfills the criteria defined in the `apply_function` the EventTriggerFunctionOperator will not complete its task like the AwaitKafkaMessageOperator, instead it will defer itself again and continue listening until the task is stopped externally for example by a timeout value of the DAG itself. Without an external termination, this task will keep deferring itself indefinitely as long as the DAG is running. View [an example DAG using this operator](https://github.com/astronomer/airflow-provider-kafka/blob/main/example_dags/listener_dag_function.py) in the provider repository.
```
Comment (kentdanas) [inline learn/airflow-kafka.md:326]:
```suggestion
The EventTriggerFunctionOperator is a [deferrable operator](https://docs.astronomer.io/learn/deferrable-operators) that waits for a specific message to be published to one or more Kafka topics, similar to the AwaitKafkaMessageOperator. Unlike the AwaitKafkaMessageOperator, after consuming a message that fulfills the criteria defined in the `apply_function` parameter, the EventTriggerFunctionOperator will continue listening until the task is stopped by an external criteria such as a timeout of the DAG itself. Without external termination, this task will stay in a deferred state indefinitely as long as the DAG is running. You can view [an example DAG using this operator](https://github.com/astronomer/airflow-provider-kafka/blob/main/example_dags/listener_dag_function.py) in the provider repository.
```

File: learn/airflow-kafka.md [hunk 2]
```diff
@@ -320,6 +321,19 @@ The AwaitKafkaMessageOperator is a [deferrable operator]((https://docs.astronome
...
+
```
Comment (kentdanas) [inline learn/airflow-kafka.md:327]: You can define the following parameters for this operator:

File: learn/airflow-kafka.md [hunk 3]
```diff
@@ -320,6 +321,19 @@ The AwaitKafkaMessageOperator is a [deferrable operator]((https://docs.astronome
...
+- `topics`: A list of topics or regex patterns to read from.
+- `apply_function`: A Python function that is applied to all messages that are read. If the function returns any data the `event_triggered_function` will run.
+- `event_triggered_function`: A Python function that will be triggered every time the operator consumes a message that cause the `apply_function` to return any data. The value returned by the apply_function is passed to the function provided to `event_triggered_function` as the first positional parameter.
```
Comment (kentdanas) [inline learn/airflow-kafka.md:330]:
```suggestion
- `event_triggered_function`: A Python function that will be triggered every time the operator consumes a message that causes the `apply_function` to return any data. The value returned by the `apply_function` is passed to the function provided to `event_triggered_function` as the first positional parameter.
```

File: learn/airflow-kafka.md [hunk 4]
```diff
@@ -320,6 +321,19 @@ The AwaitKafkaMessageOperator is a [deferrable operator]((https://docs.astronome
...
 - `xcom_push_key`: The key under which to save the returned data to XCom.
 
+### EventTriggersFunctionOperator
+
+The EventTriggerFunctionOperator is a [deferrable operator](https://docs.astronomer.io/learn/deferrable-operators) that will wait for a specific message to be published to one or more Kafka topics, similar to the AwaitKafkaMessageOperator. After consuming a message that fulfills the criteria defined in the `apply_function` the EventTriggerFunctionOperator will not complete its task like the AwaitKafkaMessageOperator, instead it will defer itself again and continue listening until the task is stopped externally for example by a timeout value of the DAG itself. Without an external termination, this task will keep deferring itself indefinitely as long as the DAG is running. View [an example DAG using this operator](https://github.com/astronomer/airflow-provider-kafka/blob/main/example_dags/listener_dag_function.py) in the provider repository.
+
+- `topics`: A list of topics or regex patterns to read from.
+- `apply_function`: A Python function that is applied to all messages that are read. If the function returns any data the `event_triggered_function` will run.
```
Comment (kentdanas) [inline learn/airflow-kafka.md:329]:
```suggestion
- `apply_function`: A Python function that is applied to all messages that are read. If the function returns any data, the `event_triggered_function` will run.
```

**Review by kentdanas — APPROVED** [review]
A couple minor edits, otherwise LGTM!

**Comment by TJaniF** [comment]
Thank you @kentdanas! @jwitz ready for you :)

File: learn/airflow-kafka.md [hunk 5]
```diff
@@ -320,6 +321,21 @@ The AwaitKafkaMessageOperator is a [deferrable operator]((https://docs.astronome
 - `poll_interval`: The amount of time in seconds that the task should wait in the deferred state.
 - `xcom_push_key`: The key under which to save the returned data to XCom.
 
+### EventTriggersFunctionOperator
+
+The EventTriggerFunctionOperator is a [deferrable operator](https://docs.astronomer.io/learn/deferrable-operators) that waits for a specific message to be published to one or more Kafka topics, similar to the AwaitKafkaMessageOperator. Unlike the AwaitKafkaMessageOperator, after consuming a message that fulfills the criteria defined in the `apply_function` parameter, the EventTriggerFunctionOperator will continue listening until the task is stopped by an external criteria such as a timeout of the DAG itself. Without external termination, this task will stay in a deferred state indefinitely as long as the DAG is running. You can view [an example DAG using this operator](https://github.com/astronomer/airflow-provider-kafka/blob/main/example_dags/listener_dag_function.py) in the provider repository.
```
Comment (jwitz) [inline learn/airflow-kafka.md:326]:
```suggestion
The EventTriggersFunctionOperator is a [deferrable operator](https://docs.astronomer.io/learn/deferrable-operators) that waits for a specific message to be published to one or more Kafka topics, similar to the AwaitKafkaMessageOperator. Unlike the AwaitKafkaMessageOperator, the EventTriggersFunctionOperator will continue listening until the task is stopped by an external criteria such as a timeout of the DAG itself. If this external criteria isn't met, this task will stay in a deferred state indefinitely as long as the DAG is running. You can view an example DAG using this operator in the [`astronomer-providers` repository](https://github.com/astronomer/airflow-provider-kafka/blob/main/example_dags/listener_dag_function.py).
```

File: learn/airflow-kafka.md [hunk 6]
```diff
@@ -320,6 +321,21 @@ The AwaitKafkaMessageOperator is a [deferrable operator]((https://docs.astronome
...
+
+You can define the following parameters for this operator:
+
+- `topics`: A list of topics or regex patterns to read from.
+- `apply_function`: A Python function that is applied to all messages that are read. If the function returns any data, the `event_triggered_function` will run.
+- `event_triggered_function`: A Python function that will be triggered every time the operator consumes a message that causes the `apply_function` to return any data. The value returned by the `apply_function` is passed to the function provided to `event_triggered_function` as the first positional parameter.
```
Comment (jwitz) [inline learn/airflow-kafka.md:332]:
```suggestion
- `event_triggered_function`: A Python function that runs every time the operator consumes a message that causes the `apply_function` to return any data. The value returned by the `apply_function` is passed to the function provided to `event_triggered_function` as the first positional parameter.
```

File: learn/airflow-kafka.md [hunk 7]
```diff
@@ -320,6 +321,21 @@ The AwaitKafkaMessageOperator is a [deferrable operator]((https://docs.astronome
...
+- `apply_function_args`: Positional arguments for the `apply_function`.
+- `apply_function_kwargs`: Keyword arguments for the `apply_function`.
+- `kafka_config`: The configuration for the Kafka client, including the connection information. For a full list of parameters please refer to the [librdkafka GitHub repository](https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md).
```
Comment (jwitz) [inline learn/airflow-kafka.md:335]:
```suggestion
- `kafka_config`: The configuration for the Kafka client, including the connection information. For a full list of parameters, see the [librdkafka GitHub repository](https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md).
```

File: learn/airflow-kafka.md [hunk 8]
```diff
@@ -320,6 +321,21 @@ The AwaitKafkaMessageOperator is a [deferrable operator]((https://docs.astronome
...
+- `poll_timeout`: The amount of time in seconds that the task should wait for a message in its active state.
+- `poll_interval`: The amonut of time in seconds that the task should wait in the deferred state.
```
Comment (jwitz) [inline learn/airflow-kafka.md:337]:
```suggestion
- `poll_interval`: The amount of time in seconds that the task should wait in the deferred state.
```

**Review by jwitz — APPROVED** [review]
LGTM!

File: learn/airflow-kafka.md (hunk 8 above)
Comment (TJaniF) [reply learn/airflow-kafka.md:337]: Second time today I made that typo. Maybe a subconscious wish for donuts? 😅

File: learn/airflow-kafka.md [hunk 9]
```diff
@@ -320,6 +321,21 @@ The AwaitKafkaMessageOperator is a [deferrable operator]((https://docs.astronome
 - `poll_interval`: The amount of time in seconds that the task should wait in the deferred state.
 - `xcom_push_key`: The key under which to save the returned data to XCom.
 
+### EventTriggersFunctionOperator
+
+The EventTriggersFunctionOperator is a [deferrable operator](https://docs.astronomer.io/learn/deferrable-operators) that waits for a specific message to be published to one or more Kafka topics, similar to the AwaitKafkaMessageOperator. Unlike the AwaitKafkaMessageOperator, the EventTriggersFunctionOperator will continue listening until the task is stopped by an external criteria such as a timeout of the DAG itself. If this external criteria isn't met, this task will stay in a deferred state indefinitely as long as the DAG is running. You can view an example DAG using this operator in the [`astronomer-providers` repository](https://github.com/astronomer/airflow-provider-kafka/blob/main/example_dags/listener_dag_function.py).
```
Comment (TJaniF) [inline learn/airflow-kafka.md:326]:
```suggestion
The EventTriggersFunctionOperator is a [deferrable operator](https://docs.astronomer.io/learn/deferrable-operators) that waits for a specific message to be published to one or more Kafka topics, similar to the AwaitKafkaMessageOperator. Unlike the AwaitKafkaMessageOperator, the EventTriggersFunctionOperator will continue listening until the task is stopped by an external criterion such as a timeout of the DAG itself. If this external criteria isn't met, this task will stay in a deferred state indefinitely as long as the DAG is running. You can view an example DAG using this operator in the [`astronomer-providers` repository](https://github.com/astronomer/airflow-provider-kafka/blob/main/example_dags/listener_dag_function.py).
```

**Pull request merged**